packages/core/src/StateNode.ts

```diff
@@ -350,7 +350,7 @@ export class StateNode<
         guardPassed =
           !guard ||
           evaluateGuard<TContext, TEvent>(guard, resolvedContext, event, state);
-      } catch (err) {
+      } catch (err: any) {
```

Comment:
it's an easy fix to actually fix those error-related locations but I'd like to take a look at how other libraries handle those cases before committing to any particular solution